legacy/core/src/main/java/com/fsck/k9/notification/SingleMessageNotificationCreator.kt

```diff
@@ -52,6 +62,12 @@ internal class SingleMessageNotificationCreator(
         notificationHelper.notify(account, notificationId, notification)
     }
 
+    private suspend fun NotificationBuilder.setAvatar(content: NotificationContent) = apply {
+        resourceProvider.avatar(content.sender)?.let {
+            setLargeIcon(IconCompat.createWithAdaptiveBitmap(it).toIcon(application))
```

Comment:
This needs to be called on minSdk 23, the current is 21. We have to bump the minSdk or update

https://github.com/thunderbird/thunderbird-android/blob/f262b0a2d0f064fe05243f06ea6f72c3c17ae5a1/legacy/ui/legacy/src/main/java/com/fsck/k9/contacts/ContactPictureLoader.kt#L59-L69

to accept a `Transformation` for `CircleCrop` for avatars.

Reply:
Use checking `Build.VERSION_CODES.M` instead as `setLargeIcon` requires API 23.